src/coreclr/tools/Common/TypeSystem/Interop/IL/MarshalHelpers.cs

```diff
@@ -797,5 +797,23 @@ public static partial class MarshalHelpers
                 return MarshallerKind.Invalid;
             }
         }
+
+        internal static bool ShouldCheckForPendingException(TargetDetails target, PInvokeMetadata metadata)
+        {
+            if (!target.IsOSX)
+                return false;
+
+            const string ObjectiveCLibrary = "/usr/lib/libobjc.dylib";
```

Comment:
I see number of examples using just `libobjc.dylib` without the full path, e.g. https://docs.microsoft.com/en-us/xamarin/ios/platform/exception-marshaling . Should this check be more relaxed?

Reply:
We were following the documentation for selectors - https://docs.microsoft.com/xamarin/ios/internals/objective-c-selectors. This specifically uses a Xamarin constant in all examples - [`ObjectiveCLibrary`](https://github.com/xamarin/xamarin-macios/blob/63ad08ed4443b659d2c58cc49dec4c4dbe478bf7/src/ObjCRuntime/Constants.cs#L9).

Reply:
Since Xamarin has a usable constant I am inclined to defer to that. Do you have a different opinion?

Reply:
I do not have an opionion. I just see it all over the place in the real world code out there: https://github.com/search?l=&p=4&q=objc_msgSend+language%3AC%23&ref=advsearch&type=Code

What is the exact condition used in existing Xamarin for the `objc_msgSend` redirection?

Reply:
The original Xamarin code used this:
```
	mono_dllmap_insert (NULL, "/usr/lib/libobjc.dylib", "objc_msgSend",            lib, "xamarin_dyn_objc_msgSend");
	mono_dllmap_insert (NULL, "/usr/lib/libobjc.dylib", "objc_msgSendSuper",       lib, "xamarin_dyn_objc_msgSendSuper");
```
so it's more of a question whether Mono's dllmap first does resolution and then matching or just matching. @CoffeeFlux?

Reply:
@jkotas I do see a few places where `objc` is just specified but the majority seem to be `LIBOBJC_DYLIB` which appears to be provided by the Xamarin code gen and points to the absolute path. I think what we have is the correct approach.

Reply:
Sounds good.

Reply:
Update documentation - https://github.com/MicrosoftDocs/xamarin-docs/pull/3305

Reply:
Just for the record: at one point we ran into problems when using just the filename. IIRC it was during a beta version of iOS, and Apple fixed their code so that using just the filename continued to work in the end, but since then we've always used the full path ourselves, and recommended it for everybody else. This is another result of this: https://github.com/xamarin/xamarin-macios/blob/e8fa7676b82ff3cd26535ec988a27cde1dc1a5d8/src/ObjCRuntime/Dlfcn.cs#L111-L118 (we expose `dlopen`, and if you don't pass a full path, we warn).